### PR TITLE
Add more versions

### PR DESCRIPTION
**Pull request by Flonja**
## Introduction
This pull request adds support for 1.16.220 to 1.12.0.
It also adds an extra configuration file to set the minimum supported protocol, for those that wish to not support legacy versions.

### Behavioural changes
- Server authoritative movement cannot be maintained in 1.12.0, due to protocol limits.
- Plugins that use custom geometry skins on players/npcs through packets should add this [line](https://github.com/Flonja/PocketMine-MP/blob/protocol/old_old_old/src/network/mcpe/convert/LegacySkinAdapter.php#L54).

## Follow-up
- The `composer.json` file should refer back to NetherGames' repositories with the correct branch names.

## Tests
~~I've tested this myself on my local machine and had 1 warning:~~
~~`[Server thread/DEBUG]: [NetworkSession: Flonja] Attempted to close inventory with network ID 255, but current is 3`~~
~~This doesn't seem like a major issue, but I'll happily fix it if it is.~~
This has been handled in this [commit](https://github.com/NetherGamesMC/PocketMine-MP/pull/194/commits/54698d55bc366c18b87e111b17e0673ebafa65ad).

Currently @Swedeachu is testing this on his practice server too and only had custom geometry issues (on 1.12) which are plugin and protocol related.

### CODE REVIEW

**Comment by Flonja** [comment]
This pull request has been ready to get reviewed for a few days. Can I get an update on the situation or why it hasn't been reviewed yet?

File: src/entity/Entity.php [hunk 1]
```diff
@@ -1474,7 +1474,7 @@ protected function sendSpawnPacket(Player $player) : void{
 			array_map(function(Attribute $attr) : NetworkAttribute{
 				return new NetworkAttribute($attr->getId(), $attr->getMinValue(), $attr->getMaxValue(), $attr->getValue(), $attr->getDefaultValue(), []);
 			}, $this->attributeMap->getAll()),
-			$this->getAllNetworkData(),
+			EntityMetadataProperties::convertProps($player->getNetworkSession()->getProtocolId(), $this->getAllNetworkData()),
```
Comment (dries-c) [inline src/entity/Entity.php]: I personally think that the AddActorPacket should have this itself

File: src/network/mcpe/convert/RuntimeBlockMapping.php [hunk 2]
```diff
@@ -90,7 +90,35 @@ private static function make() : self{
...
+			ProtocolInfo::PROTOCOL_1_14_60 => [
+				self::CANONICAL_BLOCK_STATES_PATH => '-1.14.0',
+				self::R12_TO_CURRENT_BLOCK_MAP_PATH => '-1.14.0',
+			],
+			ProtocolInfo::PROTOCOL_1_13_0 => [
+				self::CANONICAL_BLOCK_STATES_PATH => '-1.13.0',
+				self::R12_TO_CURRENT_BLOCK_MAP_PATH => '-1.13.0',
+			],
+			//ProtocolInfo::PROTOCOL_1_12_0 => [ // I tried my best, I'm pretty it's broken
+			//	self::CANONICAL_BLOCK_STATES_PATH => '-1.12.0',
```
Comment (dries-c) [inline src/network/mcpe/convert/RuntimeBlockMapping.php]: Could you remove this?

**Review by dries-c — CHANGES_REQUESTED** [review]

File: src/entity/Entity.php (hunk 1 above)
Comment (Flonja) [reply src/entity/Entity.php]: I was following the same standards as the AvailableCommandsPacket. Should I not?

Comment (dries-c) [reply src/entity/Entity.php]: Well seems like i made the mistake there then 😅

Comment (Flonja) [reply src/entity/Entity.php]: Resolved this for entity metadata, adding the same change for command parameters is out of this pr's scope

**Review by dries-c — APPROVED** [review]

**Pull request merged**